combat-trainer.lic

```diff
@@ -1997,7 +2025,186 @@ class TrainerProcess
+      'defensive combat' => @astrology_prediction_skills_defense,
+      'magic' => @astrology_prediction_skills_magic,
+      'survival' => @astrology_prediction_skills_survival,
+      'lore' => @astrology_prediction_skills_lore,
+      'future events' => 'future events'
+    }
+
+    pools.reject { |_skill, size| (size < @prediction_pool_target) }
+         .each_key do |skill|
+      break if DRSkill.getxp('Astrology') > 30
+      align(skillset_to_pool[skill])
+    end
+  end 
+
+  def check_heavens
+    @equipment_manager.empty_hands
+    vis_bodies = visible_bodies
+    night = vis_bodies.find { |body| body['constellation'] }
+    best_eye_data = vis_bodies
+                    .select { |data| @have_telescope || !data['telescope'] }
+                    .max_by { |data| [data['pools'].values.compact.size, data['circle']] }
+    echo(" best_eye_data = #{best_eye_data}") if UserVars.astrology_debug
+    waitrt?
+    if @have_telescope
+      things_to_try = @constellations.select do |data|
+        data['telescope'] &&
+          data['circle'] <= DRStats.circle &&
+          data['circle'] > best_eye_data['circle'] &&
+          (night || !data['constellation']) &&
+          data['pools'].values.compact.size > best_eye_data['pools'].values.compact.size
+      end
+      things_to_try << best_eye_data
+      things_to_try.sort! { |data| data['circle'] }.reverse!
+      echo(" things_to_try = #{things_to_try}") if UserVars.astrology_debug
+      things_to_try.find do |data|
+        result = nil
+        until ['Roundtime', 'You peer aimlessly through your telescope'].include?(result)
+          return check_heavens if Flags['bad-search'] == 'is foiled by the daylight'
+          next if Flags['bad-search'] == 'turns up fruitless'
+          result = observe(data['name'])
+        end
+        result == 'Roundtime'
+      end
+    else
+      until observe(best_eye_data['name'])
+        if Flags['bad-search']
+          check_heavens
+          return
+        end
+      end
+    end
+    pause 2
+    waitrt?
+  end  
+  
+  def observe(body)
+    Flags.add('bad-search', 'is foiled by the (daylight|darkness)', 'turns up fruitless')
+    if @have_telescope
+      DRC.bput('get telescope', 'you get', 'You are already', "That can't be picked up")
+      case DRC.bput("center telescope on #{body}", 'You put your eye', 'open it to make any use of it', 'The pain is too much')
+      when /open it/
+        DRC.bput('open my telescope', 'extend your telescope')
+        return observe(body)
+      when /The pain is too much/
+        return
+        return observe(body)
+      end
+      waitrt?
+      pause 1
+      fput("retreat")
+      fput("retreat")
+      result = DRC.bput('peer telescope', 'You peer aimlessly through your telescope', 'open it to make any use of it', 'You see nothing regarding the future', 'The pain is too much', 'Roundtime')
+      case result
+      when /The pain is too much/
```

Comment:
This means you cannot perform astrology. I would consider adding some logic to handle that here. I could see users getting stuck here

Reply:
done